src/game.cpp

```diff
@@ -9144,7 +9144,14 @@ bool game::walk_move( const tripoint &dest_loc )
     }
     u.set_underwater( false );
 
-    if( !shifting_furniture && !pushing && !prompt_dangerous_tile( dest_loc ) ) {
+    if( !shifting_furniture && !pushing && ( is_dangerous_tile( dest_loc ) &&
+            !u.movement_mode_is( CMM_RUN ) ) ) {
+        if( is_dangerous_tile( dest_loc ) && !u.movement_mode_is( CMM_RUN ) ) {
```

Comment:
This was already checked in the lines above. It won't have changed. You don't need to check it again.

Reply:
Good catch, thanks